client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get, noop } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+
+/**
+ * Dispatches a request to save particular onboarding settings on a site
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const saveJetpackOnboardingSettings = ( { dispatch, getState }, action ) => {
+	const { settings, siteId } = action;
+	const state = getState();
+	const token = get( state.jetpackOnboarding.credentials, [ siteId, 'token' ], null );
+	const jpUser = get( state.jetpackOnboarding.credentials, [ siteId, 'userEmail' ], null );
+
+	return dispatch(
+		http(
+			{
+				apiVersion: '1.1',
+				method: 'POST',
+				path: '/jetpack-blogs/' + siteId + '/rest-api/',
+				body: {
+					path: '/jetpack/v4/settings/',
+					body: JSON.stringify( {
+						onboarding: {
+							...settings,
+							token,
+							jpUser,
+						},
+					} ),
```

Comment:
Just to make sure I'm getting this right -- Is this the endpoint and request shape we'll be using for all the other JPO settings too? E.g. what about contact page creation -- what would the request look like for that?

Reply:
> Just to make sure I'm getting this right -- Is this the endpoint and request shape we'll be using for all the other JPO settings too?

Yes, the plan is to use the same shape. 

>  E.g. what about contact page creation -- what would the request look like for that?

For creating a contact page, we need to add a `addContactForm: true` parameter to the request, which we can do by calling `saveJetpackOnboardingSettings()` with `{ addContactForm: true }` in the `settings` argument.

If you want to see all parameters that are currently accepted in the JPO request, you can refer to:
https://github.com/Automattic/jetpack/blob/master/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L941

Reply:
> If you want to see all parameters that are currently accepted in the JPO request, you can refer to:
https://github.com/Automattic/jetpack/blob/master/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L941

Thanks, was going to ask! 😁